### PR TITLE
fix(document-handling): refactor the Objectmapper supplier and make it private so only copy can be created

### PR DESCRIPTION
**Pull request by mathias-vandaele**
## Description

Done during the hacking session with the team

### CODE REVIEW

File: secret-providers/gcp-secret-provider/src/main/java/io/camunda/connector/runtime/cloud/GcpSecretManagerSecretProvider.java [hunk 1]
```diff
@@ -40,34 +40,29 @@
 
 public class GcpSecretManagerSecretProvider implements SecretProvider {
 
+  /** Secrets used as fallback if SecretProvider is loaded via SPI */
+  public static final String SECRETS_PROJECT_ENV_NAME = "SECRETS_PROJECT_ID";
+
+  public static final String SECRETS_PREFIX_ENV_NAME = "SECRETS_PREFIX";
```
Comment (johnBgood) [inline secret-providers/gcp-secret-provider/src/main/java/io/camunda/connector/runtime/cloud/GcpSecretManagerSecretProvider.java:46]: is there any changes in this class? If not let's revert the changes (re ordering I guess)

File: connector-sdk/test/src/main/java/io/camunda/connector/test/outbound/OutboundConnectorContextBuilder.java [hunk 2]
```diff
@@ -44,7 +45,13 @@ public class OutboundConnectorContextBuilder {
   protected Map<String, Object> variables;
   protected DocumentFactory documentFactory =
       new DocumentFactoryImpl(InMemoryDocumentStore.INSTANCE);
-  private ObjectMapper objectMapper = ConnectorsObjectMapperSupplier.getCopy();
+  private ObjectMapper objectMapper =
+      ConnectorsObjectMapperSupplier.getCopy()
```
Comment (johnBgood) [inline connector-sdk/test/src/main/java/io/camunda/connector/test/outbound/OutboundConnectorContextBuilder.java:49]: should be the other `getCopy(.....)` I guess

File: connector-sdk/test/src/main/java/io/camunda/connector/test/inbound/InboundConnectorContextBuilder.java [hunk 3]
```diff
@@ -51,7 +51,7 @@ public class InboundConnectorContextBuilder {
   protected InboundConnectorDefinition definition;
   protected ValidationProvider validationProvider;
 
-  protected ObjectMapper objectMapper = ConnectorsObjectMapperSupplier.DEFAULT_MAPPER;
+  protected ObjectMapper objectMapper = ConnectorsObjectMapperSupplier.getCopy();
```
Comment (johnBgood) [inline connector-sdk/test/src/main/java/io/camunda/connector/test/inbound/InboundConnectorContextBuilder.java:54]: should we also plug the document capability here?

File: connector-runtime/spring-boot-starter-camunda-connectors/src/main/java/io/camunda/connector/runtime/OutboundConnectorsAutoConfiguration.java [hunk 4]
```diff
@@ -163,8 +163,7 @@ public ConsoleSecretApiClient consoleSecretApiClient(CamundaClientProperties cli
   @Bean
   @ConditionalOnMissingBean
   public ObjectMapper objectMapper(DocumentFactory documentFactory) {
-    ConnectorsObjectMapperSupplier.registerDocumentModule(
-        documentFactory, DocumentModuleSettings.create());
+    ConnectorsObjectMapperSupplier.getCopy(documentFactory, DocumentModuleSettings.create());
```
Comment (johnBgood) [inline connector-runtime/spring-boot-starter-camunda-connectors/src/main/java/io/camunda/connector/runtime/OutboundConnectorsAutoConfiguration.java:166]: we should return this line not the one below

**Review by johnBgood — COMMENTED** [review]
Just a few comments :)

File: secret-providers/gcp-secret-provider/src/main/java/io/camunda/connector/runtime/cloud/GcpSecretManagerSecretProvider.java (hunk 1 above)
Comment (mathias-vandaele) [reply secret-providers/gcp-secret-provider/src/main/java/io/camunda/connector/runtime/cloud/GcpSecretManagerSecretProvider.java:46]: Yes thanks to spotless 🤔

**Review by johnBgood — APPROVED** [review]

**Review by chillleader — APPROVED** [review]
🔥

File: connectors-e2e-test/connectors-e2e-test-rabbitmq/src/test/java/io/camunda/connector/e2e/RabbitMqInboundStartEventTests.java [hunk 5]
```diff
@@ -131,9 +131,10 @@ public void uriAuthenticationReceiveMessageTest() throws Exception {
 
   private void assertIntermediateCatchEventUsingModel(BpmnModelInstance model) throws Exception {
     Object expectedJsonResponse =
-        ConnectorsObjectMapperSupplier.DEFAULT_MAPPER.readValue(
-            "{\"message\":{\"consumerTag\":\"myConsumerTag\",\"body\":{\"foo\": {\"bar\": \"barValue\"}},\"properties\":{}}}",
-            Object.class);
+        ConnectorsObjectMapperSupplier.getCopy()
```
Comment (sbuettner) [inline connectors-e2e-test/connectors-e2e-test-rabbitmq/src/test/java/io/camunda/connector/e2e/RabbitMqInboundStartEventTests.java:134]: Can we reuse it between the tests?
